File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/criticalandeffortdriventasks.rb
module Asposetasksjava
  module CriticalAndEffortDrivenTasks
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		# Parse through all the collected tasks
        tasks = collector.getTasks()
        i = 0
        while i < tasks.size()
        	task = tasks.get(i)
	        puts str_ed = task.get(tsk.IS_EFFORT_DRIVEN) != nil ? "EffortDriven" : "Non-EffortDriven"
	 	    puts str_crit = task.get(tsk.IS_CRITICAL) != nil ? "Critical" : "Non-Critical"
            i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/CriticalAndEffortDrivenTasks.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\ChildTasksCollector as ChildTasksCollector;
use com\aspose\tasks\TaskUtils as TaskUtils;
use com\aspose\tasks\Tsk as Tsk;
class CriticalAndEffortDrivenTasks{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

            # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tsk = new Tsk();

		# Parse through all the collected tasks
        $tasks = $collector->getTasks();
        $i = 0;
        while ($i < sizeof($tasks)) {
            $task = $tasks -> get($i);
	        print $str_ed = $task -> get($tsk -> IS_EFFORT_DRIVEN) != null ? "EffortDriven" : "Non-EffortDriven".PHP_EOL;
	 	    print $str_crit = $task -> get($tsk -> IS_CRITICAL) != null ? "Critical" : "Non-Critical".PHP_EOL;
            $i += 1;
		}

    }

}